src/main/java/org/baldogru/imagegallery/service/validator/FileExtensionValidator.java

```diff
@@ -1,23 +1,37 @@
 package org.baldogru.imagegallery.service.validator;
 
+import org.apache.commons.io.FilenameUtils;
+import org.baldogru.imagegallery.constants.ValidationStatus;
 import org.baldogru.imagegallery.model.dto.RequestData;
 import org.baldogru.imagegallery.model.dto.ValidationResult;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.Optional;
 
 @Service
 final class FileExtensionValidator implements SpecializedValidator {
 
-    FileExtensionValidator() {}
+    FileExtensionValidator() {
+    }
 
     @Value("'${image.allowed.extensions}'.split(',')")
     private List<String> allowedImageExtensions;
 
     @Override
     public ValidationResult validate(RequestData requestData) {
-        //Polecam skorzystac z tego gowna z apache commons-> FilenameUtils.getExtension()
-        return null;
+        String fileName = requestData.getImageFile().getName();
+        String fileExtension = FilenameUtils.getExtension(fileName);
+        Optional<String> validatedFile = allowedImageExtensions.stream().filter(allowedExtension -> allowedExtension.equals(fileExtension)).findFirst();
+        if (validatedFile.isPresent()) {
```

Comment:
Cały ten kod od tego miejsca do końca metody da się dużo ładniej zrobić. Sam wymyslisz czy mam wrzucić przykład? :)

Reply:
Hmm oprócz rozbicia na jakąś inną metodę, to nie przychodzi mi nic do głowy. 
Oczywiście zapomniałem znowu sprwadzić, czy czasem requestData nie jest nullem tak samo jak requestData.getImageFile()